File: 14-LongestCommonPrefix/14-LongestCommonPrefix.rb
# Last updated: 5/30/2025, 7:35:20 PM
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    prefix = ""
    length = strs.map { |s| s.length }.min
    0.upto(length - 1) do |i|
        char = strs[0][i]
        strs.each { |s| return prefix if s[i] != char }
        prefix += char
    end
    prefix
end

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 5/30/2025, 7:35:27 PM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:          
        pre = ""
        for i in range(len(strs[0])):
            for word in strs:
                if i == len(word) or word[i]!=strs[0][i]:
                    return pre
            pre += strs[0][i]
        return pre
